src/pdf_parser/pdf_utils/ocr.py

```diff
@@ -0,0 +1,480 @@
+import concurrent.futures
+import io
+from collections import defaultdict
+from typing import Optional, Tuple, List, Union
+
+import numpy as np
+from PIL.PpmImagePlugin import PpmImageFile
+from google.cloud import vision
+from google.cloud.vision import types
+from google.cloud.vision_v1.types import BoundingPoly
+from layoutparser import TextBlock, Rectangle, Layout
+from layoutparser.ocr import TesseractAgent, GCVAgent
+from pydantic import BaseModel as PydanticBaseModel
+from shapely.geometry import Polygon
+from google.protobuf.pyext._message import RepeatedCompositeContainer
+
+from src.base import PDFTextBlock
+from src.pdf_parser.pdf_utils.disambiguate_layout import lp_coords_to_shapely_polygon
+
+
+class BaseModel(PydanticBaseModel):
```

Comment:
Why are we re-defining this class?

Reply:
Good point. Moved to higher level module.